src/main/java/net/dv8tion/jda/internal/entities/TextChannelImpl.java

```diff
@@ -357,6 +358,22 @@ public MessageAction sendMessage(@Nonnull Message msg)
         return TextChannel.super.sendMessage(msg);
     }
 
+    @Nonnull
+    @Override
+    public MessageAction sendFile(@Nonnull File file, @Nonnull String fileName, @Nonnull AttachmentOption... options)
+    {
+        checkPermission(Permission.MESSAGE_READ);
+        checkPermission(Permission.MESSAGE_WRITE);
+        checkPermission(Permission.MESSAGE_ATTACH_FILES);
+
+        final long allowedFileSize = Math.max(getJDA().getSelfUser().getAllowedFileSize(), getGuild().getMaxFileSize());
+        Checks.check(file.length() <= allowedFileSize,
+            "File may not exceed the maximum file length of %d bytes!", allowedFileSize);
```

Comment:
You have to null check the file here too. Is it even possible for `selfUser.getAllowedFileSize()` to be smaller than `guild.getMaxFileSize()`?

Reply:
> 
> 
>  Is it even possible for `selfUser.getAllowedFileSize()` to be smaller than `guild.getMaxFileSize()`?

What do you mean? Assuming the self user has the limit of 8 MiB and Guild's Boost Tier is 3 (100 MiB limit), the answer is yes.

Reply:
I meant higher, you use Math.max here but is it even necessary?

Reply:
I understand now.